### ISSUE TITLE
MissingJavadocMethodCheck: top-level public method without Javadoc not flagged in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20218]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/javadoc/missingjavadocmethod.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="MissingJavadocMethod"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputMissingJavadocMethodCompact.java <<'EOF'
public void processOrder(String id) {
    System.out.println(id);
}
void main() { }
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml InputMissingJavadocMethodCompact.java

Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**

MissingJavadocMethodCheck must flag a top-level public method without Javadoc in a JEP 512 compact source file, exactly as it does for the same method inside a normal class. The check's `shouldCheck` calls `ScopeUtil.getSurroundingScope(ast)` to determine the enclosing class's access level; for a top-level method in a compact source file there is no enclosing `CLASS_DEF`/`INTERFACE_DEF`/`ENUM_DEF`/`ANNOTATION_DEF`/`RECORD_DEF`, so `getSurroundingScope` returns `Optional.empty()` and `shouldCheck` returns false. The method is never validated regardless of its access modifier or the configured scope.

Expected: a violation `Missing a Javadoc comment.` on `processOrder`, identical to what the same method inside a normal class produces with the default configuration.

**Comment by Carbon14-48** [comment]
i'm on this

**Comment by vivek-0509** [comment]
@Carbon14-48 Hey, please pick up some other issue for now. This check/fix will touch the same part of the codebase that I’m already working on in one of my PRs, so it may lead to unnecessary conflicts. Also, I have a few questions regarding this issue that I’ll post once I start working on it, so please avoid picking this one up for now. Feel free to take any other issues where I haven’t left a comment. Thanks!

**Comment by Carbon14-48** [comment]
> @Carbon14-48 Hey, please pick up some other issue for now. This check/fix will touch the same part of the codebase that I’m already working on in one of my PRs, so it may lead to unnecessary conflicts. Also, I have a few questions regarding this issue that I’ll post once I start working on it, so please avoid picking this one up for now. Feel free to take any other issues where I haven’t left a comment. Thanks!
> 

Okayyyyyyy !!! Thanks for the Early reply !

**Comment by vivek-0509** [comment]
@stoyanK7 @romani @mahfouz72 
After we made the fix in this [PR](https://github.com/checkstyle/checkstyle/pull/20326) this issue is fixed..
because original root cause (getSurroundingScope returning Optional.empty()) no longer applies the ifx made it return Scope.PACKAGE for the implicit class, which is semantically correct: a JEP 512 implicit class is final, in the unnamed package, and can't be referenced by name, so a public top-level method's effective visibility is capped at package level. MissingJavadocMethodCheck already gates on effective visibility (surroundingScope.isIn(scope)), so it deliberately does not flag a public method inside a package-private class at scope=public. The compact source file is the same situation, so not flagging it is consistent and correct.

We can close this issue, the compact source file input for this is already added....

**Comment by stoyanK7** [comment]
Thanks @vivek-0509
